omgeo/services/__init__.py

```diff
@@ -920,3 +920,87 @@ def _geocode(self, pq):
             c.geoservice = self.__class__.__name__
             returned_candidates.append(c)
         return returned_candidates
+
+class Mapzen(GeocodeService):
+    """
+    Class to geocode using `Mapzen search service
+    <https://mapzen.com/projects/search>`_.
+
+    Settings used by the Mapzen GeocodeService object include:
+     * api_key --  The API key used to access search service.
+
+    """
+    _wkid = 4326
+
+    DEFAULT_PREPROCESSORS = [ReplaceRangeWithNumber()] # 766-68 Any St. -> 766 Any St.
+
+    def __init__(self, preprocessors=None, postprocessors=None, settings=None):
+        if settings.has_key('api_version'):
+            self._api_version = 'v' + str(settings['api_version'])
+        else:
+            self._api_version = 'v1'
+
+        if settings.has_key('instance_url'):
+            self._base_url = settings['instance_url']
+        else:
+            self._base_url = 'https://search.mapzen.com'
+
+        self._default_endpoint = self._base_url + '/' + \
```

Comment:
Using `urlparse.urljoin` may be a little safer here. If the client passes `instance_url` with a trailing slash, we won't end up with a double slash between the base and the version.

https://docs.python.org/2/library/urlparse.html#urlparse.urljoin

```python
In [1]: from urlparse import urljoin
In [2]: urljoin('http://mine.com/', '/v9.9/search')
Out[2]: 'http://mine.com/v9.9/search'
In [3]: urljoin('http://mine.com', 'v9.9/search')
Out[3]: 'http://mine.com/v9.9/search'